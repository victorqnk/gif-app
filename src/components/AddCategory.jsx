import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
  const [value, setValue] = useState('')

  const onChange = ({ target }) => setValue(target.value)

  const onSubmit = (event) => {
    event.preventDefault()
    if (value.trim().length <= 3) return
    onNewCategory(value.trim())
    setValue('')
  }

  return (
    <form onSubmit={onSubmit}>
      <input
        onChange={onChange}
        placeholder='Search gif'
        type='text'
        value={value}
      />
      <button onClick={onSubmit}>add category</button>
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func,
}
