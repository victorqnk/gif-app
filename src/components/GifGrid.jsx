import { GifItem } from './GifItem'
import { useFetchGifs } from '../hooks/useFetchGifs'
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category)

  return (
    <>
      <h4>{category}</h4>
      {isLoading && <h6>LOADING...</h6>}

      <div className='card-grid'>
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
