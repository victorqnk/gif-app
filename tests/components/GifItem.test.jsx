import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('GifItem component', () => {
  const title = 'Saitama'
  const url = 'https://one-punch.com/saitama.jpg'

  it('should match the snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  it('should show correct image with alt text', () => {
    render(<GifItem title={title} url={url} />)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  it("should show component's title", () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})
