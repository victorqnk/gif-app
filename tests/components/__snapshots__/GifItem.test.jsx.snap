// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem component should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <small>
      Saitama
    </small>
  </div>
</div>
`;
